simplified-app-shared/src/main/java/org/nypl/simplified/app/AnnotationsManager.kt

```diff
@@ -0,0 +1,600 @@
+import org.slf4j.LoggerFactory
+import java.io.IOException
+
+/**
+ * Performs work relevant to syncing OPDS Annotations.
+ * Current features for books:
+ * 1 - Current reading position
+ * 2 - Bookmarks saved by the user
+ */
+class AnnotationsManager(private val libraryAccount: Account,
+                         private val credentials: AccountCredentials,
+                         private val context: Context)
+{
+  private val requestQueue = Volley.newRequestQueue(context)
+
+  private companion object {
+    val LOG = LoggerFactory.getLogger(AnnotationsManager::class.java)!!
+  }
+
+  /**
+   * Queries the server, and based on a set of conditions, decides whether
+   * or not to inform the user about "Syncing" in an Alert Dialog, where
+   * the user can choose to dismiss/ignore, or turn it on for the first time.
+   * @param account The particular library for the permission status request.
+   * @param completion Asynchronous handler for the user's decision, or error.
+   */
+  fun requestServerSyncPermissionStatus(account: AccountsControllerType,
+                                        completion: (enableSync: Boolean) -> Unit)
+  {
+    if (!syncIsPossible(account)) {
+      LOG.debug("Account does not satisfy conditions for sync setting request.")
+      completion(false)
+      return
+    }
+
+    //TODO I think this condition is either wrong or out of date.. think about it some more during testing
+    if (Simplified.getSharedPrefs().getBoolean("userHasSeenFirstTimeSyncMessage") == true &&
+        Simplified.getSharedPrefs().getBoolean("syncPermissionGranted", libraryAccount.id) == false) {
+      completion(false)
+      return
+    }
+
+    syncPermissionStatusUriRequest { initialized, syncIsPermitted ->
+      if (initialized && syncIsPermitted) {
+        Simplified.getSharedPrefs().putBoolean("userHasSeenFirstTimeSyncMessage", true)
+        LOG.debug("Sync has already been enabled on the server. Enable here as well.")
+        completion(true)
+      } else if (!initialized && !Simplified.getSharedPrefs().getBoolean("userHasSeenFirstTimeSyncMessage")) {
+        LOG.debug("Sync has never been initialized for the patron. Proceeding with opt-in alert.")
+        presentFirstTimeSyncAlertDialog(completion)
+      } else {
+        LOG.debug("Sync has been initialized and not permitted. Continuing with sync disabled.")
+        completion(false)
+      }
+    }
+  }
+
+  private fun syncPermissionStatusUriRequest(completion: (initialized: Boolean, syncIsPermitted: Boolean) -> Unit)
+  {
+    val catalogUriString = libraryAccount.catalogUrl ?: null
+    val baseUri = if (catalogUriString != null) Uri.parse(catalogUriString) else null
+    val permissionRequestUri = if (baseUri != null) Uri.withAppendedPath(baseUri, "patrons/me/") else null
+
+    if (permissionRequestUri == null) {
+      LOG.debug("Could not create Annotations URL from Main Feed URL. Abandoning attempt to retrieve sync setting.")
+      return
```

Comment:
shouldn't we call the completion block here with an error? otherwise this would just cause the user to sit on a page.

Reply:
Yeah you're right. I've updated this. I've made a note to go back and refactor the two usages of this since one caller requires the completion for correct execution, and the other does not.